scylla/src/statement/query.rs

```diff
@@ -28,6 +28,18 @@ impl Query {
         }
     }
 
+    pub fn with_page_size(contents: String, page_size: i32) -> Self {
+        Self {
+            contents,
+            page_size: Some(page_size),
+            consistency: Default::default(),
+            serial_consistency: None,
+            is_idempotent: false,
+            retry_policy: None,
+            tracing: false,
+        }
+    }
```

Comment:
You can make this function much shorter:

```rust
pub fn with_page_size(contents: String, page_size: i32) -> Self {
    Self {
        page_size: Some(page_size),
        ..Self::new(contents)
    }
}
```

Reply:
On the other hand, maybe we could convert it to a builder-style method? It would take `self` by value and return a new query:

```rust
pub fn with_page_size(mut self, page_size: i32) -> Self {
    self.page_size = page_size;
    self
}
```

Usage:

```rust
// Before:
Query::with_page_size(contents, page_size)

// After:
Query::new(contents).with_page_size(page_size)
```

Slightly longer, but with this pattern you can add more `with_xyz` methods and chain them.